src/statsmodels/contrasts.jl

```diff
@@ -0,0 +1,143 @@
+## Specify contrasts for coding categorical data in model matrix. Contrast types
+## are constructed with data and other options, creating a subtype of
+## AbstractContrast. AbstractContrast provides the interface for creating modle
+## matrix columns and coefficient names
+##
+## ModelFrame will hold a Dict{Symbol, T<:AbstractContrast} that maps column
+## names to contrasts.
+##
+## ModelMatrix will check this dict when evaluating terms, falling back to a
+## default for any categorical data without a specified contrast.
+
+abstract AbstractContrast
+
+termnames(term::Symbol, col::Any, contrast::AbstractContrast) =
+    ["$term - $name" for name in contrast.termnames]
```

Comment:
Right, adding a separator isn't a bad idea. `"$term: $name"` is good, though we need to think about how interactions will appear. For example: `x: A&y` would naturally be read as  `x: (A&y)`, instead of `(x: A)&y`; it would be even worse with interactions between categorical variables. Removing the space after the semicolon might be enough. (The bikeshedding session is open!)